File: server-laravel-single/app/Policies/TaskPolicy.php
<?php

namespace App\Policies;

use Rhino\Policies\ResourcePolicy;
use Illuminate\Contracts\Auth\Authenticatable;

/**
 * Single-tenant TaskPolicy. Any authenticated user may CRUD; TaskScope enforces
 * ownership (via the parent Project). All attributes are exposed.
 */
class TaskPolicy extends ResourcePolicy
{
    public function permittedAttributesForShow(?Authenticatable $user): array
    {
        return $user ? ['*'] : [];
    }

    public function permittedAttributesForCreate(?Authenticatable $user): array
    {
        return $user
            ? ['title', 'description', 'status', 'priority', 'estimated_hours', 'due_date', 'project_id', 'assignee_id']
            : [];
    }

    public function permittedAttributesForUpdate(?Authenticatable $user): array
    {
        return $this->permittedAttributesForCreate($user);
    }
}

File: server-rails-hybrid/app/policies/task_policy.rb
# frozen_string_literal: true

class TaskPolicy < Rhino::ResourcePolicy
  def permitted_attributes_for_show(user)
    if has_role?(user, "owner") || has_role?(user, "admin") || has_role?(user, "manager")
      ["*"]
    elsif has_role?(user, "member") || has_role?(user, "viewer")
      %w[id title description status priority due_date project_id assignee_id]
    else
      []
    end
  end

  def hidden_attributes_for_show(user)
    if has_role?(user, "member") || has_role?(user, "viewer")
      %w[estimated_hours]
    else
      []
    end
  end

  def permitted_attributes_for_create(user)
    if has_role?(user, "owner") || has_role?(user, "admin") || has_role?(user, "manager")
      %w[title description status priority estimated_hours due_date project_id assignee_id]
    else
      []
    end
  end

  def permitted_attributes_for_update(user)
    if has_role?(user, "owner") || has_role?(user, "admin") || has_role?(user, "manager")
      %w[title description status priority estimated_hours due_date project_id assignee_id]
    elsif has_role?(user, "member")
      %w[status description]
    else
      []
    end
  end
end
